src/KernelAbstractions.jl

```diff
@@ -532,40 +532,64 @@ get_backend(::Array) = CPU()
 Adapt.adapt_storage(::CPU, a::Array) = a
 
 """
-    allocate(::Backend, Type, dims...)::AbstractArray
+    allocate(::Backend, Type, dims...; unified=false)::AbstractArray
 
-Allocate a storage array appropriate for the computational backend.
+Allocate a storage array appropriate for the computational backend. `unified`
+allocates an array using unified memory if the backend supports it. Use
+[`supports_unified`](@ref) to determine whether it is supported by a backend.
 
 !!! note
     Backend implementations **must** implement `allocate(::NewBackend, T, dims::Tuple)`
 """
-allocate(backend::Backend, T::Type, dims...) = allocate(backend, T, dims)
-allocate(backend::Backend, T::Type, dims::Tuple) = throw(MethodError(allocate, (backend, T, dims)))
+allocate(backend::Backend, T::Type, dims...; kwargs...) = allocate(backend, T, dims; kwargs...)
+function allocate(backend::Backend, T::Type, dims::Tuple; unified::Union{Nothing, Bool} = nothing)
+    return if isnothing(unified)
+        throw(MethodError(allocate, (backend, T, dims)))
+    elseif unified
+        throw(ArgumentError("`$(typeof(backend))` either does not support unified memory or it has not yet defined `allocate(backend::$backend, T::Type, dims::Tuple; unified::Bool)`"))
```

Comment:
We can use the query to check?

Reply:
I had the following:
```suggestion
        if supports_unified(backend)
            throw(ArgumentError("`$(typeof(backend))` states support for unified memory, but has not defined `allocate(backend::$(typeof(backend)), T::Type, dims::Tuple; unified::Bool)`. Pleas "))
        else
            throw(ArgumentError("`$(typeof(backend))` does not support unified memory. If you believe it does, please open a github issue requesting adding support for it."))
        end
```

but this feels really complicated for no reason. I think 

```suggestion
        throw(ArgumentError("`$(typeof(backend))` does not support unified memory. If you believe it should, please open a github issue requesting adding support for it."))
```
would work since whether a device supports it or not, it's true that the julia KA backend for it does not in the situation where a user is encountering this error.